src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -3,37 +3,55 @@
 #include <cos_debug.h>
 #include <cos_types.h>
 #include <llprint.h>
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+#include <sl.h>
+#include <llboot.h>
+#include <res_spec.h>
 
 #define UNDEF_SYMBS 64
 
 /* Assembly function for sinv from new component */
-extern void *__inv_test_entry(int a, int b, int c);
+extern void *llboot_entry_inv(int a, int b, int c);
+extern int num_cobj;
+extern int resmgr_spdid;
+extern int root_spdid;
 
 struct cobj_header *hs[MAX_NUM_SPDS + 1];
 
 /* The booter uses this to keep track of each comp */
 struct comp_cap_info {
 	struct cos_compinfo *compinfo;
+	struct cos_defcompinfo *defci;
```

Comment:
I don't really follow why we have a defcompinfo inlined into this, but I'll with-hold judgement for now.

Reply:
I need to remove these and add wrappers to get these struct pointers.

Reply:
got rid of all redundant pointers in an update to this PR, including global "boot_info" pointer.